### ISSUE TITLE
Why is 'invalid@email' valid email address? It isn't valid email address

### ISSUE THREAD

**Issue by obivan86** [Respect/Validation#931]

**Comment by williamespindola** [comment]
You can provide some pice of code that you are using to validate? Because this is a invalid emails if you take a look on [email rule tests](https://github.com/Respect/Validation/blob/1.1/tests/unit/Rules/EmailTest.php#L130-L146)

**Comment by henriquemoody** [comment]
The fact is that `invalid@email` is a valid email. You can have an username `invalid` in a host named `email`, and that would be valid.

You're probably using `egulias/email-validator`, which we use to validate emails according to a couple RFCs. It's quite complicated to validate emails because there are way to many possibilities. If you do not use `egulias/email-validator` you would still have some problems with a couple of emails, because you will have some false positives by using a REGEX or `filter_var($input, FILTER_VALIDATE_EMAIL)` (which is what we use).

The way I do recommend you to validate emails is to send an email and then check for the confirmation, that's the safer way.

**Comment by nickl-** [comment]
I agree with @henriquemoody

[RFC5322 Section 3.4.1](https://tools.ietf.org/html/rfc5322#section-3.4.1) defines the `addr-spec` as

```
local-part "@" domain
```

Even though TLDs (top level domains ie. com, org) don't traditionally serve as domain names there is nothing in the DNS specification preventing TLDs for being used for this purpose and `user@com` would make a valid e-mail if the appropriate DNS configurations were applied to the `com` record. 

As @henriquemoody points out domain is also commonly used to refer to hostname and on most linux systems the address `root@localhost` is a valid e-mail address for the system administrator. Only useful if you can see `localhost` but no less a valid e-mail.

As @henriquemoody  also pointed out, the only true way to know if it is really a valid and useable address is to send a mail and see if it gets a delivery failure report. This will work in most cases but it still depends on the configuration of the host mail service, as there is nothing forcing us to respond with failures. The host could simply decide to delete undeliverable e-mails and you would never know that it was invalid or it could land in the users spam folder equally undeliverable in most cases. You might think that checking if the host is valid, ie. by pinging the host, is a way to verify the host but this will be a mistake as the email records for a domain is not related to the lookup records and there is nothing forcing a valid e-mail domain to also have a valid response to pings or visa versa. These of course are all beyond the scope of our e-mail validation which only validates if a supplied string has the appearance of a valid e-mail address.

For our purpose `invalid@email` does have the appearance of a valid e-mail address.

Closing: works as implemented.
